packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts

```diff
@@ -61,6 +61,38 @@ export function isAzureProject(azureSettings: AzureSolutionSettings | undefined)
   return azureSettings !== undefined && HostTypeOptionAzure.id === azureSettings.hostType;
 }
 
+export function isBotProject(azureSettings: AzureSolutionSettings | undefined): boolean {
+  return (
+    azureSettings !== undefined &&
+    (azureSettings.capabilities?.includes(BotOptionItem.id) ||
+      azureSettings.capabilities?.includes(MessageExtensionItem.id))
+  );
+}
+
+export interface IBotTroubleShootMessage {
```

Comment:
It is a good idea to have this object to hold all message.

Reply:
The interface name could be BotTroubleShootMessage and we don't need to add I for it because Iinterface is only the convention for C#

Reply:
Agree, thanks for reminding. I will align this with TypeScript.